src/main/java/com/btkelly/gnag/reporters/BaseReporter.java

```diff
@@ -95,8 +99,22 @@ private T getReport(Project project) throws JAXBException {
         }
     }
 
-    private java.io.File getReportFile(Project project) {
-        return new java.io.File(project.getProjectDir(), getReportFilePath());
+    @Nullable
+    private java.io.File getReportFile(Project project) throws FileNotFoundException {
```

Comment:
I actually removed the Nullable annotation since the function was now throwing exceptions to indicate failure states. The only valid return value will be non-null as before. The new exceptions thrown are handled further up the method call chain, the same as other exceptions - they are logged, but don't mark the build as failed. This might be something to investigate changing?